### ISSUE TITLE
FriendsOfGalaxy no longer operational, link directly to upstream contribs instead

### ISSUE THREAD

**Issue by grnassar** [borger/scoop-galaxy-integrations#18]
This repo references https://github.com/Mixaill/awesome-gog-galaxy for the list of repos available; it should be noted that in the majority of cases, that "official" list links to the original developer's plugin repo, not the FriendsOfGalaxy fork.

This is relevant because FriendsOfGalaxy has been inactive and plugins have gone unmaintained for over four months now.  This has broken a number of plugins due to not pulling from the original developer's repo as they were regularly doing prior to January of this year.  (For reference, please see: https://github.com/gogcom/galaxy-integrations-python-api/issues/194)

Recommending that the scoop bucket point to the original upstream developers that FoG forks from, rather than the FoG fork, to match what Mixaill/awesome-gog-galaxy is doing -- which will also fix the broken plugin issues with a number of repos.  (If this is already the case, then the bug request can be considered to refer to the editing of README.md to reflect the link change, as it presents only the FriendsOfGalaxy forks under the "Integrations Available in this Bucket" list.)

**Comment by borger** [comment]
Hello @grnassar ,

I apologize for the delay in replying, I was looking into this, specifically, as you mentioned on your second link, Rockstar and Uplay integrations (as the others seem to be working fine, despite not getting any updates for months). So looking at their upstream projects it appears they are worse, for instance, Rockstar upstream repo https://github.com/tylerbrawl/Galaxy-Plugin-Rockstar latest release is from 2019! Yes, we could just use the 'src' folder and remap it in scoop, but I wonder what's going on with FriendsOfGalaxy...

Did you try using the upstream projects? did they work? Besides Rockstar and Uplay, did you find any other integration with issues?

Thanks for reporting.

**Comment by GoAwayNow** [comment]
There's a fork of the Ubisoft Connect plugin that's functional (and will hopefully stay maintained).

[GOG-Nebula/galaxy-integration-uplay](https://github.com/GOG-Nebula/galaxy-integration-uplay)

**Comment by grnassar** [comment]
> I apologize for the delay in replying, I was looking into this, specifically, as you mentioned on your second link, Rockstar and Uplay integrations (as the others seem to be working fine, despite not getting any updates for months). So looking at their upstream projects it appears they are worse, for instance, Rockstar upstream repo https://github.com/tylerbrawl/Galaxy-Plugin-Rockstar latest release is from 2019! Yes, we could just use the 'src' folder and remap it in scoop, but I wonder what's going on with FriendsOfGalaxy...
> 
> Did you try using the upstream projects? did they work? Besides Rockstar and Uplay, did you find any other integration with issues?

Yes, I used the upstream projects (as does anyone that directly uses the Mixaill/awesome-gog-galaxy list does, as I mentioned - highly recommended!).  To take the Rockstar integration that you mention as an example: As you may have noticed from the README in that repo noting a particular issue (which I created!) and from there looking at their PRs in the FriendsOfGalaxy downstream, one can surmise that the lack of a release is actually *because* of their introduction into the FriendsOfGalaxy auto-updater.  The auto-updater works via PRs from the upstream source, not via published releases.

You can see this, as that issue indicates, over at https://github.com/FriendsOfGalaxy/galaxy-integration-rockstar/pulls -- the pull is still there, requesting a pull from tylerbrawl/Galaxy-Plugin-Rockstar into FriendsOfGalaxy/galaxy-integration-rockstar, and the creator of the PR is FriendsOfGalaxyBot.  Remove the default "open" filter, and you can see that FriendsOfGalaxyBot has created pull requests for every (unpublished) upstream release since Nov 27, 2019.  (Before then, there are pull requests back to Oct 29, 2019, done by the FriendsOfGalaxy account.)  This matches your observation of the last published release being Nov 25, 2019.

(If one wants to dig further into it, it seems that the FriendsOfGalaxy auto-updater, while it was working, basically polled the file src/version.py for the __version__ variable, and if it had changed from the prior pull it created a PR with that version number in its title.  Creating actual github releases would therefore be optional on the developer's part, and arguably could be intentionally avoided to encourage use via auto-updater rather than the repo version which would probably lead to fewer issues headaches due to end users installing incorrectly.  I imagine @tylerbrawl could confirm my guess or correct any errors, however.)

The tl;dr here is: yes, the upstream works, and of course this must be the case and it's been actively maintained seeing as you can see a steady stream of accepted PRs into the FriendsOfGalaxy downstream from 2019 straight until April of 2021 (again, by removing the "open" filter from the PR list).

I have found this to be the case for every such listed repo in Mixaill/awesome-gog-galaxy that I have looked at.  I cannot claim that I have exhaustively checked every single such listing, but I can say that I do use an awful lot of these plugins, and every time I've had a problem with an auto-updated repo in Galaxy and the Mixaill/ list points to the upstream rather than FriendsOfGalaxy, directly using the upstream as it implicitly recommends has fixed the problem.  (Humble Bundle is another example of that same thing; the upstream maintainer has kept up changes even though GOG has not, and using the upstream directly fixed all my issues.)  That, plus the fact that it's been nearly two months since I posted the request here and as you can see GOG has *still* not moved to fix this problem, is why I recommend said remedy.

---

There is a related issue here, that I avoided mentioning as it probably belongs in a different issue but is mentioned in @GoAwayNow 's earlier comment so for the sake of completeness I'll address it here.  That is as follows:

As I said above, using the upstream repo has fixed the issues "every time I've had a problem with an auto-updated repo in Galaxy **and the Mixaill/ list points to the upstream rather than FriendsOfGalaxy**".  I have encountered two cases where there is an issue with a repo, but looking at its entry in the Mixaill list one finds that it points to the FriendsOfGalaxy repo rather than the upstream directly.  One of those cases, as the commenter indicated, was with the Ubisoft plugin.  (The other was with the Origin plugin.)  Again, kudos to the maintainer of the Mixaill list -- they correctly noticed that the listed "upstream" for the FriendsOfGalaxy repo wasn't actually the upstream; looking at the commit logs, one can see that whoever FriendsOfGalaxy was (probably a former employee at this point), they were actually (and very unofficially, I'm sure) maintaining the Ubisoft (and Origin) plugin themselves!

There are three or four forks of that repo at this point which have fixed the (remarkably simple) problem with the Ubisoft plugin.  I'm sure if there was anybody over at FriendsOfGalaxy still actually doing their job, they would've selected one of these forks to be the upstream repo by now -- of course, that doesn't seem like it's going to happen any time soon, and certainly the Galaxy 2.x developers don't seem to be in a big hurry to decouple the app from the FriendsOfGalaxy repo itself for some mysterious reason.  None seem to have any further track record of maintenance to the plugin; ironically there is a fourth fork that does seem to have added new functionality but hasn't fixed the original bug yet!  Suffice it to say, the state of that plugin is a bit of a mess.

The tl;dr to *this* second section is: Using the same links that Mixaill/ uses fixes the original problems this issue was created for; it does not fix all problems, namely ones where upstream has not been maintained, but it seems in every case so far it correctly identifies and links to the most current and best-maintained version of the "in-app official" upstream source (even when it's FriendsOfGalaxy themselves!).  They have clearly done a lot of legwork keeping that maintained, and I think it would be useful to reflect that work in the Scoop bucket.  My only recommendation as far as the second point re: the Ubi plugin is "if one is to do something about that one plugin, it probably needs to be done in another issue." :)

**Comment by grnassar** [comment]
I suppose that was a bit too tl;dr -- I wanted to give background on the topic, since you'd mentioned you'd been looking into the issue.
If so, the tl;dr version is: No, the Rockstar upstream definitely *is* better (even before their recent rolling of a "proper" non-dev release); no, these two are not the only platforms affected (Humble is as well, for one); and most importantly there are a number of other broken plugins with community fixes that will never work in their current state in the bucket -- fixable either by including them in this bucket, which is probably a lot of work that would potentially duplicate work done by this bucket's intended reference ("if you are looking for more integrations to be manually installed or to be added to this bucket, check the full list at [awesome-gog-galaxy](https://github.com/Mixaill/awesome-gog-galaxy)"), or more easily just by conforming to the reference, fixing some problems immediately and giving people a place to go to submit fixes for other problems.  Once those changes are submitted to the Mixaill list and hopefully approved for addition, simple conformance to that reference fixes the problem here too without making you have to figure out which forks are good or bad.

**Comment by ABaumher** [comment]
Coming here from the Steam Integration. Since early March, Steam implemented a new way of authentication, and it completely broke the integration. See https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/159 for all the people who have brought it up. 

We're still developing a fix, so there is no module to point to yet, but when it is is done, FriendsOfGalaxy being inactive means we will have no way to update the integration you reference. We will let you know when it is done, but in the meantime, is there anything we need to do to be considered for your Steam integration source?
As mentioned by @GoAwayNow above, GalaxyNebula has active support and could take over for FriendsOfGalaxy. We (the people working on the fix) could do a pull request there and contribute to upkeep if you'd prefer.
